lib/Dialect/TritonNvidiaGPU/Transforms/WSPipeline.cpp

```diff
@@ -642,6 +644,26 @@ void buildAsyncComm(const DenseMap<Operation *, SmallVector<Channel *>> &map,
     agentsPC.insert(agentsPC.end(), agentP.begin(), agentP.end());
     agentsPC.insert(agentsPC.end(), agentC.begin(), agentC.end());
   };
+
+  auto getValidDot = [&](const SmallVector<Channel *> &block) -> Operation * {
+    Operation *headConsumer = block.front()->dstOp;
+    if (block.size() == 2 &&
+        isa<triton::DotOp>(*headConsumer->getUsers().begin()) &&
+        headConsumer->getParentOfType<scf::ForOp>()) {
+      auto dotOp = cast<triton::DotOp>(*headConsumer->getUsers().begin());
+      auto dot = dotOp.getResult();
+      auto resTy = dot.getType().dyn_cast<RankedTensorType>();
+      auto CArg = dotOp.getOperand(2).dyn_cast<BlockArgument>();
+      // Don't pipeline dots that depend on ops other than scf.yield and scf.for
```

Comment:
Could we add some abstract reason why we should not pipeline in other cases?

Reply:
yeah. I added some comments.